resources/js/components/fieldtypes/Fieldtype.vue

```diff
@@ -26,20 +26,89 @@ export default {
     },
 
     methods: {
-        update(value) {
-            this.$emit('input', value);
+
+        update(input) {
+            this.$emit("input", this.isInputEvent(input) ? input.target.value : input);
+            this.updateCursorPosition(input);
         },
 
-        updateDebounced: _.debounce(function (value) {
-            this.update(value);
+        updateDebounced: _.debounce(function (input) {
+            this.update(input);
         }, 150),
 
         updateMeta(value) {
             this.$emit('meta-updated', value);
-        }
+        },
+
+        blurEvent() {
+            Statamic.user.cursor = null;
+            this.$emit('blur');
+        },
+
+        updateCursorPosition(input) {
+            let position = null;
+
+            if (this.isInputEvent(input) || this.isPointerEvent(input)) {
+                position = {
+                    start: input.target.selectionStart,
+                    end: input.target.selectionStart
+                };
+            }
+
+            if (this.isKeyboardEvent(input)) {
+                // If using the keydown event, the selection position is so to say always one number behind.
+                let start = input.target.selectionStart;
+
+                // Substract one position if moving to the left, if not already on position zero.
+                if (input.code === "ArrowLeft" && start !== 0) {
+                    start--;
+                }
+
+                // Add one position if moving to the right, if not already on the last position.
+                if (input.code === "ArrowRight" && input.target.value.length > start) {
+                    start++;
+                }
+
+                // Move position to start if pressing arrow up.
+                if (input.code === "ArrowUp") {
+                    start = 0;
+                }
+
+                // Move position to last position if pressing arrow up.
+                if (input.code === "ArrowDown") {
+                    start = input.target.value.length;
+                }
+
+                position = {
+                  start: start,
+                  end: start // Is the same as start
+                };
+            }
```

Comment:
Why not avoid all this and just do

```js
position = {
  start: input.target.selectionStart,
  end: input.target.selectionEnd
};
```

?

Reply:
Oh because of it being one step behind, I think.

Reply:
If you change the events from `keydown` to `keyup`, it will be correct. The position will have changed by the time the key is released.

Reply:
That has been my first approach, which didn't work somehow. So you are looking at my workaround for that problem. 

Let me try that approach one more time tomorrow, to check if I did miss something.

Reply:
You'll need to have the Text.vue component also emit the keyup events. At the moment it's only emitting keydown.

Reply:
Yes, anything else would not make sense :-)

Reply:
You were right. I did miss the keyup event inside Text.vue